pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -0,0 +1,173 @@
+package defaultevictor
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/descheduler/pkg/apis/componentconfig"
+	nodeutil "sigs.k8s.io/descheduler/pkg/descheduler/node"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
+	"sigs.k8s.io/descheduler/pkg/utils"
+)
+
+const (
+	PluginName            = "DefaultEvictor"
+	evictPodAnnotationKey = "descheduler.alpha.kubernetes.io/evict"
+)
+
+var _ framework.Plugin = &DefaultEvictor{}
+var _ framework.EvictorPlugin = &DefaultEvictor{}
+
+type constraint func(pod *v1.Pod) error
+
+// DefaultEvictor is the first EvictorPlugin, which defines the default extension points of the
+// pre-baked evictor that is shipped.
+// Even though we name this plugin DefaultEvictor, it does not actually evict anything,
+// This plugin is only meant to customize other actions (extension points) of the evictor,
+// like filtering, sorting, and other ones that might be relevant in the future
+type DefaultEvictor struct {
+	args        *componentconfig.DefaultEvictorArgs
+	constraints []constraint
+}
+
+// IsPodEvictableBasedOnPriority checks if the given pod is evictable based on priority resolved from pod Spec.
+func IsPodEvictableBasedOnPriority(pod *v1.Pod, priority int32) bool {
+	return pod.Spec.Priority == nil || *pod.Spec.Priority < priority
+}
+
+// HaveEvictAnnotation checks if the pod have evict annotation
+func HaveEvictAnnotation(pod *v1.Pod) bool {
+	_, found := pod.ObjectMeta.Annotations[evictPodAnnotationKey]
+	return found
+}
+
+type DefaultEvictorOptions struct {
+	Priority      *int32
+	NodeFit       bool
+	LabelSelector labels.Selector
+}
+
+// New builds plugin from its arguments while passing a handle
+func New(args runtime.Object, nodeIndexer podutil.GetPodsAssignedToNodeFunc) (framework.EvictorPlugin, error) {
```

Comment:
All plugin constructors are expected to comply with `func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error)` signature.